### DISCUSSION TITLE
Make text chat more noticeable

### DISCUSSION THREAD

**Discussion by evilfactory** [FakeFishGames/Barotrauma#7882] [Ideas and suggestions]
Text chat is often ignored by people that use microphone, to combat this, i propose a few solutions:

- Make the text chat GUI more noticeable instead of something that disappears after 10 seconds, i know there's a button that makes the text chat stay always open, but most people don't know about this, so an easy solution is to make it always open by default?

- Add text boxes that appear above the player when they send something in chat, similar to ss13 (best solution in my opinion)
![image](https://user-images.githubusercontent.com/36804725/146813041-302e914a-38ec-4f10-b3cc-45cf2de00e81.png)

**Comment by gerakis** [comment]
Time to get a microphone.

**Reply by ghost** [reply]
There are price limits involved and background noise can cause issues, and some people do not want to reveal their voice, such as myself. I have a fairly good microphone but choose not to use it.

**Reply by gerakis** [reply]
You can buy a decent microphone for 2 dollars off ebay. The second issue sounds like mental problems. Text chat is wildly inferior to voice no matter how noticeable you make it.

**Reply by ghost** [reply]
Not saying that text chat is better than voice, but you do not understand that not everyone wants to use a mic, and some cannot. Those 3 reasons are not an exhaustive list.

**Reply by mygamingaccount** [reply]
You don't understand. Your voice is what gives personality to the in-game avatar. Without it, you're just an expendable array of pixels.
Players won't connect and form bonds, friendships with crewmates, if they can't be bothered to spend 2$ for what is essential piece of gear in an online multiplayer game.

**Reply by evilfactory** [reply]
I disagree. Sure, voice chat gives a little bit of personality, but does it make a difference for me? No. I actually prefer text chat, it's way easier to read 5 people sending messages than trying to listen to 5 people talking in radio at the same time, and in my experience, rounds where everyone was using voice chat were way more chaotic and fast paced than rounds that only a few people were using voice chat.

Personally it usually boils down to: am i playing with 5 players or less? if yes, then i consider using voice chat, if not, i don't use it, simple.

**Comment by mygamingaccount** [comment]
Remove text chat from the game entirely

**Reply by gerakis** [reply]
I would agree with removing text. I already autoban anyone without a mic.

**Reply by Kafkatrap** [reply]
> Remove text chat from the game entirely

And remove the ability to send text commands to your subs components and everything else that comes with having a text chat?

Like this: 
https://steamcommunity.com/sharedfiles/filedetails/?id=1780299560

I don't think you know what you're asking for. This has to be a troll suggestion.

**Comment by Ek-Videogames** [comment]
**A few points about voip**
Can be pretty annoying to be unable to hear important orders the first time due to:
- other sound effects in game drowning it out _(Loud machines like oxygen-gen or fabricators/deconstructors are really bad for this)_
- people talking at the same time or being at wildly different mic volume levels
- poor signal quality
- or just generally being too distracted doing something in-game to focus the mental energy to parse what anybody's saying in any of the above difficult situations.

So even in voicechat lobby you might often use text chat in order to convey important messages or things that need to be remembered.
Especially stuff like shopping lists, or repair alerts, etc.

**Comment by Working-Joe** [comment]
Just expending on the emergency button system so that it's more noticable might work. Some animated pulsing outline for the more urgent things especially. Along with being able to highlight chat somehow. Small mockup as an example:
![image](https://user-images.githubusercontent.com/50582661/177147880-11ce03ce-a1c6-4ab6-8d88-a957ecb71c5a.png)

Could even be an option to allow pressing the button multiple times to make the icon cycle through emergencies. The same system may also be useful for job orders, to highlight a periscope or work site, or perhaps even the world pos of an aiming target via the sonar monitor.

**Reply by mygamingaccount** [reply]
If the emergency report system is already being touched, I'd recommend a button that is more generic for non-emergency use.
Have you had the following interaction happen:
A) Anyone saw the Incendium I bought?
B) It's in the engine room
A) Where's that?
B) Let me ping it!
C) INTRUDERS IN ENGINE ROOM??
